### DISCUSSION TITLE
Using Google Sheets to maintain data

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#1248] [Help] [answered]
Hi is it posable to use google sheets to do calculations with the data?
can I have users input data via Budibase's interface. Budibase will then record the data send it to google sheets, there I will have google sheets do some calculations and Budibase will then read and import the new calculated columns where I can use them for graphs and display the info generated by google sheets, in Budibase.

any help / workarounds would be helpful.
Thank you

**Comment by shogunpurple** [comment]
Hi @AndrewJRamay - right now there is no direct google sheets integration for budibase, but it should be done over the coming months.

The best way to achieve this at the moment is to use a tool like zapier or integromat to interact with the webhook feature in budibase.
- Set up an integromat/zapier automation to write to google sheets
- Set up a webhook in integromat/zapier to write back to budibase

**Reply by KernOil** [reply]
Than you very much

**Reply by KernOil** [reply]
do you have any tutorials on webhooks working with / setting up in budibase?

**Comment by joebudi** [comment, marked as answer]
Hey Andrew. This is not possible now, but Google sheet integration is on the immediate roadmap.

How complex are the calculations?

**Reply by KernOil** [reply]
this would be a huge value to directly connect to sheets. they are not too complicated. its just calculating API gravity and a few other basic oil and gas metrics. I just need user input then a calculated column based off user input (+,- / * by a constant ) nothing too crazy.
how far out do you think your integration is. I was looking at your release timeline and you guys can move!

**Reply by joebudi** [reply]
Hey. @shogunpurple is best to answer that. If you can pull your data into Budibase, you could use Handlebars to do the calculations.

Is there a particurlar reason you need Google sheets? I'm thinking it might be a possibility to download the data from your sheet as a CSV, upload it to Budibase, then user handlebars to run basic calculations over the data in your design panel.

https://youtu.be/Ff_ceH31XAw 

The video above might give you idea of how handlebarsjs is used in the builder.

**Reply by KernOil** [reply]
ok this might work my concern is will those calculated values then be captured in the database as a new line? i need to capture before and after calculation numbers.  
the reason for sheets is because I'm not good a coding and I know excel very well.  it seams like a solution for the limited coders.

**Reply by joebudi** [reply]
Understandable. Google sheets integration will come in the next few weeks - it might be best to wait until then.

**Reply by hendrikjap** [reply]
Hi @joebudi , is google sheets integration coming?

**Reply by shogunpurple** [reply]
Hi @ely2727 - the google sheets integration will be shipped with the January release in a few weeks.
